src/web/components/Core/PagingToolHelper.spec.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  getPageNumberOptions,
+  getRowsPerPageOptions,
+  getShowingRowsText,
+} from './PagingToolHelper';
+
+describe('test paging tool helper functions', () => {
+  describe('check that correct rows per page options are returned', () => {
+    it('should return at least one rows per page option', () => {
+      expect(getRowsPerPageOptions(3)).toEqual([{ name: '10', id: 10 }]);
+      expect(getRowsPerPageOptions(1)).toEqual([{ name: '10', id: 10 }]);
+      expect(getRowsPerPageOptions(10)).toEqual([{ name: '10', id: 10 }]);
+    });
+    it('should return all rows per page options', () => {
+      expect(getRowsPerPageOptions(250)).toEqual([
+        { name: '10', id: 10 },
+        { name: '25', id: 25 },
+        { name: '50', id: 50 },
+        { name: '100', id: 100 },
+        { name: '250', id: 250 },
+      ]);
+      expect(getRowsPerPageOptions(125)).toEqual([
+        { name: '10', id: 10 },
+        { name: '25', id: 25 },
+        { name: '50', id: 50 },
+        { name: '100', id: 100 },
+        { name: '250', id: 250 },
+      ]);
+      expect(getRowsPerPageOptions(500)).toEqual([
+        { name: '10', id: 10 },
+        { name: '25', id: 25 },
+        { name: '50', id: 50 },
+        { name: '100', id: 100 },
+        { name: '250', id: 250 },
+      ]);
+    });
+  });
+
+  describe('check that correct page numbers are returned', () => {
```

Comment:
Similar in vein to my other comments, there's quite a few magic numbers and repetition here that makes it hard to follow - perhaps we could try extracting some constants, parameterisem etc.?